packages/ui/src/components/InputFields/SmartInputField/CountryCodeSelector/index.tsx

```diff
@@ -1,40 +1,62 @@
+import type { Nullable } from '@silverhand/essentials';
 import classNames from 'classnames';
-import type { ChangeEventHandler, ForwardedRef } from 'react';
-import { useMemo, forwardRef } from 'react';
+import type { ForwardedRef } from 'react';
+import { useCallback, useState, useMemo, forwardRef } from 'react';
 
 import DownArrowIcon from '@/assets/icons/arrow-down.svg';
+import { onKeyDownHandler } from '@/utils/a11y';
 import { getCountryList, getDefaultCountryCallingCode } from '@/utils/country-code';
 
+import CountryCodeDropdown from './CountryCodeDropdown';
 import * as styles from './index.module.scss';
 
 type Props = {
   className?: string;
   value?: string;
-  onChange?: ChangeEventHandler<HTMLSelectElement>;
+  inputRef?: Nullable<HTMLInputElement>;
+  onChange?: (value: string) => void;
 };
 
 const CountryCodeSelector = (
-  { className, value, onChange }: Props,
+  { className, value, inputRef, onChange }: Props,
   ref: ForwardedRef<HTMLDivElement>
 ) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
   const countryList = useMemo(getCountryList, []);
   const defaultCountCode = useMemo(getDefaultCountryCallingCode, []);
 
+  const showDropDown = useCallback(() => {
+    setIsDropdownOpen(true);
+  }, []);
```

Comment:
Simply
```suggestion
  const showDropDown = () => {
    setIsDropdownOpen(true);
  };
```
Since the `setIsDropdownOpen` won't mutate.

Reply:
Updated. 
P.S. the newly assigned function will mutate even `setIsDropdownOpen` won't.  Remove the useCallback as a mutated eventHandler is acceptable.